network/hnswrapper/hnsv2wrapperwithtimeout.go

```diff
@@ -342,6 +342,29 @@ func (h Hnsv2wrapperwithtimeout) ListEndpointsOfNetwork(networkId string) ([]hcn
 	}
 }
 
+// NOTE see assumptions of hnsv2wrapperfake.ListEndpointsQuery
+func (h Hnsv2wrapperwithtimeout) ListEndpointsQuery(query hcn.HostComputeQuery) ([]hcn.HostComputeEndpoint, error) {
+	r := make(chan ListEndpointsFuncResult)
```

Comment:
this leaks goroutines when they try to send to the channel after the timeout has fired. fix is to make this a buffered chan of size 1.